vxquery-core/src/main/javacc/xquery-grammar.jj

```diff
@@ -1743,17 +1744,54 @@ NameTestNode Wildcard():
     }
 }
 
-ASTNode FilterExpr()   :
+ASTNode PostfixExpr()   :
 {
     ASTNode expr;
-    List<ASTNode> pList;
+    ASTNode arguments;
+    List<ASTNode> argList=new ArrayList<ASTNode>();
 }
 {
-    expr = PrimaryExpr() pList = PredicateList() {
-        FilterExprNode fe = new FilterExprNode(expr.getSourceLocation());
-        fe.setExpr(expr);
-        fe.setPredicates(pList);
-        return fe;
+    expr = PrimaryExpr() (arguments = Predicate() {
+        argList.add(arguments);
+        } | arguments = Arguments() {
+            argList.add(arguments);
+        })*{
```

Comment:
Indentation seems off. Make sure opening and closing braces line up on each line.